var/spack/repos/builtin/packages/extrae/package.py

```diff
@@ -35,54 +35,55 @@ class Extrae(Package):
     # https://www.gnu.org/software/gettext/FAQ.html#integrating_undefined
-    depends_on("gettext")
-    depends_on("binutils+libiberty")
 
-    def install(self, spec, prefix):
-        if 'openmpi' in spec:
-            mpi = spec['openmpi']
-        elif 'mpich' in spec:
-            mpi = spec['mpich']
-        elif 'mvapich2' in spec:
-            mpi = spec['mvapich2']
+    build_directory = 'spack-build'
 
-        extra_config_args = []
+    variant('dyninst', default=False, description="Use dyninst for dynamic code installation")
+    depends_on('dyninst@:9', when='+dyninst')
 
-        # This was added due to configure failure
-        # https://www.gnu.org/software/gettext/FAQ.html#integrating_undefined
-        extra_config_args.append('LDFLAGS=-lintl')
+    variant('papi', default=True, description="Use PAPI to collect performance counters")
+    depends_on('papi', when='+papi')
+
+    def configure_args(self):
+        spec = self.spec
+        args = ["--with-mpi=%s" % spec['mpi'].prefix,
+                "--with-unwind=%s" % spec['libunwind'].prefix,
+                "--with-boost=%s" % spec['boost'].prefix,
+                "--with-dwarf=%s" % spec['libdwarf'].prefix,
+                "--with-elf=%s" % spec['elf'].prefix,
+                "--with-xml-prefix=%s" % spec['libxml2'].prefix,
+                "--with-binutils=%s" % spec['binutils'].prefix]
+
+        args += (["--with-papi=%s" % spec['papi'].prefix]
+                 if '+papi' in self.spec else
+                 ["--without-papi"])
+
+        args += (["--with-dyninst=%s" % spec['dyninst'].prefix]
+                 if '+dyninst' in self.spec else
+                 ["--without-dyninst"])
 
         if spec.satisfies("^dyninst@9.3.0:"):
             make.add_default_arg('CXXFLAGS=-std=c++11')
-            extra_config_args.append('CXXFLAGS=-std=c++11')
+            args.append('CXXFLAGS=-std=c++11')
 
-        configure("--prefix=%s" % prefix,
-                  "--with-mpi=%s" % mpi.prefix,
-                  "--with-unwind=%s" % spec['libunwind'].prefix,
-                  "--with-dyninst=%s" % spec['dyninst'].prefix,
-                  "--with-boost=%s" % spec['boost'].prefix,
-                  "--with-dwarf=%s" % spec['libdwarf'].prefix,
-                  "--with-papi=%s" % spec['papi'].prefix,
-                  "--with-dyninst-headers=%s" % spec[
-                      'dyninst'].prefix.include,
-                  "--with-elf=%s" % spec['elf'].prefix,
-                  "--with-xml-prefix=%s" % spec['libxml2'].prefix,
-                  "--with-binutils=%s" % spec['binutils'].prefix,
-                  "--with-dyninst-libs=%s" % spec['dyninst'].prefix.lib,
-                  *extra_config_args)
+        # This was added due to configure failure
+        # https://www.gnu.org/software/gettext/FAQ.html#integrating_undefined
+        args.append('LDFLAGS=-lintl')
 
-        make()
-        make("install", parallel=False)
```

Comment:
Can you confirm that both 3.7.1 and 3.4.1 install in parallel? If not, we'll have to add something like:
```python
def install(self, spec, prefix):
    with working_dir(self.build_directory):
        make('install', parallel=False)
```
This can also be made conditional on the version number.

Reply:
Works for me, but checking the extrae changelog I see that there is a parallel make patch only made it into the tree for v 3.7.0+